src/common/components/repositories-home/index.js

```diff
@@ -89,9 +92,14 @@ class RepositoriesHome extends Component {
 
   render() {
     const { snaps } = this.props;
-    const hasSnaps = (snaps.snaps && Object.keys(snaps.snaps).length > 0);
-    // show spinner until we know if user has any enabled repos
-    return (snaps.isFetching && !hasSnaps)
+    const hasSnaps = (snaps.snaps && snaps.snaps.length > 0);
+
+    // show spinner if snaps data was not yet fetched (snaps list is empty)
+    // (to avoid spinner during polling we are not checking `success` or `isFetching`
+    //
+    // when snaps are loaded and user doesn't have any, they will be redirected
+    // to select repositories (so spinner won't be showing endlessly)
+    return !hasSnaps
```

Comment:
In case of 'My repos' we don't care about `isFetching` or success.

If we don't know if user has any repos (initial state) we show spinner.
If we know they don't have any, we redirect to select repositories.

Otherwise we render repositories list. There is no spinner shown during polling.